src/common/types/time.cpp

```diff
@@ -273,6 +273,14 @@ dtime_t Time::FromTime(int32_t hour, int32_t minute, int32_t second, int32_t mic
 	return dtime_t(result);
 }
 
+bool Time::TryFromTime(int32_t hour, int32_t minute, int32_t second, int32_t microseconds, dtime_t &result) {
+	if (!Time::IsValidTime(hour, minute, second, microseconds)) {
```

Comment:
The parser guarantees this to be true. There is an asymmetry with dates because dates need to be checked for month length, leap years and so on. I'd merge my PR (it's only two files).

Reply:
What time ranges are valid depends on the context, so this is not always correct (it is not correct when using a format string for example). So let's just remove this.